### DISCUSSION TITLE
Barrier Function Material of order 6 with defined parameters

### DISCUSSION THREAD

**Discussion by MScKarl** [idaholab/moose#20557] [Q&A General] [answered]
Hi!
I am completely new to MOOSE as a simulation tool, and I have a question related to a project I am working on. 

Using the BarrierFunctionMaterial with g_order high, I am only getting a fourth order polynomial, is there a way for me to define this for a sixth order polynomial and also define the different parameters? Otherwise is there some other kind of work around I can use to solve this problem?

All help and explanation is appreciated. Thanks beforehand. 

Best regards 
Karl

**Comment by dschwen** [comment, marked as answer]
You could just use a DerivativeParsedMaterial to provide the barrier function.

**Reply by MScKarl** [reply]
Thank you for your answer. If a could ask a follow-up question, it would be very much appreciated.

Using the BarrierFunctionMaterial with g_order high, (g(η) = η^2(1−η ^2) ^2). Is it possible, in a straightforward way, to add coefficients in front of the η-term, where one of the terms needs to be space-depended?

Best regards

**Reply by GiudGiud** [reply]
See my answer in your post. Daniel's answer also applies to this imo
